nixos/modules/services/matrix/draupnir.nix

```diff
@@ -0,0 +1,267 @@
+      description = lib.mdDoc ''
+        File containing the access token for Draupnir's Matrix account.
+      '';
+    };
+
+    pantalaimon = mkOption {
+      description = lib.mdDoc ''
+        `pantalaimon` options (enables E2E Encryption support).
+
+        This will create a `pantalaimon` instance with the name "draupnir".
+      '';
+      default = { };
+      type = types.submodule {
+        options = {
+          enable = mkEnableOption (lib.mdDoc ''
+            pantalaimon, in order to enable E2EE support.
+            If `true`, accessToken is ignored and the username/password below will be
+            used instead. The access token of the bot will be stored in the dataPath.
+          '');
+
+          username = mkOption {
+            type = types.str;
+            description = lib.mdDoc ''
+              Account name on the Matrix homeserver.
+            '';
+          };
+
+          passwordFile = mkOption {
+            type = with types; nullOr path;
+            default = null;
+            description = lib.mdDoc ''
+              File containing the password for the Matrix account.
+            '';
+          };
+
+          options = mkOption {
+            type = types.submodule (import ./pantalaimon-options.nix);
+            default = { };
+            description = lib.mdDoc ''
+              Pass through additional options to the `pantalaimon` service.
+            '';
+          };
+        };
+      };
+    };
+    verboseLogging = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Whether Draupnir should log a lot more messages in the room,
+        mainly involves "all-OK" messages, and debugging messages for when draupnir checks bans in a room.
+      '';
+    };
+
+    dataPath = mkOption {
+      type = types.path;
+      default = "/var/lib/draupnir";
+      description = lib.mdDoc ''
+        The directory the bot should store various bits of information in.
+      '';
+    };
+
+    managementRoom = mkOption {
+      type = types.str;
+      default = "#moderators:example.org";
+      description = lib.mdDoc ''
+        The room ID or alias where moderators can use the bot's functionality.
+
+        The bot has no access controls, so anyone in this room can use the bot - secure this room!
+
+        Warning: When using a room alias, make sure the alias used is on the local homeserver!
+        This prevents an issue where the control room becomes undefined when the alias can't be resolved.
+      '';
+    };
+
+    protectedRooms = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      example = literalExpression ''
+        [
+          "https://matrix.to/#/#yourroom:example.org"
+          "https://matrix.to/#/#anotherroom:example.org"
+        ]
+      '';
+      description = lib.mdDoc ''
+        A list of rooms to protect (matrix.to URLs).
+        These can also be configured interactively.
+      '';
+    };
```

Comment:
Pretty sure these options are now covered by `settings`, but as they're very relevant they could be moved as options into settings.